### ISSUE TITLE
[feature_request] display all visited cmake files in CMake project outline

### ISSUE THREAD

**Issue by DoDoENT** [vector-of-bool/vscode-cmake-tools#655]
### Brief Issue Summary

I recently migrated from using QtCreator to VSCode with this extension for my daily development due to VSCode having much better support for different languages (no more need to have 3 editors for editing python, C++ and Groovy code in the same project) and also because VSCode's C++ debugger is simply awesome - much better than the one provided by QtCreator.

However, I am missing one feature: in QtCreator when I configured my CMake project, its CMake outline displayed all CMake files that were visited (traversed) during the CMake configuration process. On the other hand, the CMake outline in VSCode displays only CMakeLists.txt files for each of my projects but does not display all *.cmake files that were included (either directly or transitively) from the main CMakeLists.txt. Adding support for that would make VSCode much more usable, especially because quick open feature would then be able to open CMake files that were used, even though if they are not directly part of the workspace, but are included from CMAKE_MODULE_PATH folders (as is case when using Conan package manager and Conan package that provides some CMake utilities).

### Expected:

CMake outline view should display all *.cmake files that were visited during the CMake configuration process.

### Apparent Behavior:

CMake outline view displays only source files for each project and main CMakeLists.txt file.

### Platform and Versions

- **Operating System**: MacOS 10.14.3
- **CMake Version**: 3.13.4
- **VSCode Version**: 1.31.1
- **CMake Tools Extension Version**: 1.1.3
- **Compiler/Toolchain**: Xcode 10.1

**Comment by dcourtois** [comment]
Hi,

Could you provide a small example ? In my current projects I don't see anything missing in the CMake view of VSCode compared to the one in QtCreator.

Ideally a screenshot of the view in QtCreator and the corresponding view in VSCode to highlight what's missing would be great :)

**Comment by DoDoENT** [comment]
Hi!
[Here](https://github.com/vector-of-bool/vscode-cmake-tools/files/3049704/SampleCMakeProject.zip) is a sample project that demonstrates the issue. You will need [conan package panager](https://conan.io) to run it.

Here is how it looks like in VSCode:
<img width="1792" alt="Screenshot 2019-04-05 at 23 27 07 CEST" src="https://user-images.githubusercontent.com/158844/55657436-80736f00-57fa-11e9-9fe1-848f63391420.png">

and here is how it looks like in QtCreator:
<img width="1792" alt="Screenshot 2019-04-05 at 23 27 20 CEST" src="https://user-images.githubusercontent.com/158844/55657452-8c5f3100-57fa-11e9-82a1-b35c2a62f3b5.png">

Note that files `Project.cmake`, `Project.srcs.cmake`, `ProjectTest.cmake` and `ProjectTest.srcs.cmake` do not appear in VSCode outline, even though those files are executed by CMake.

Even better, notice the statement `include( FindGTest )` (added just to demonstrate the issue, even though this example does not use it). VSCode is not aware of the `FindGTest.cmake` (which comes from the `gtest` conan package and pressing Cmd+P and searching for `FindGTest.cmake` does not result with anything. On the other hand, QtCreator nicely displays full path to `FindGTest.cmake` (it shows nicely that it came from different path) and pressing Cmd+K and searching for `FindGTest.cmake` immediately finds it.

When having lots of cmake scripts coming from various external paths (mostly from my private conan packages) it is quite annoying that VSCode CMake outline is not aware of them, even though cmake normally executed them.

I hope this now better explains the issue and gives you the way of reproducing it.

**Comment by dcourtois** [comment]
Yup, thanks ! In my projects I'm using find_package, and the files did not appear in QtCreator, so I didn't understand what you meant :) If I can find a bit of time, I'll see if it can be done ! (I'm not the owner of this addon, just trying to help with some occasional pull requests :))
